lib/ansible/modules/cloud/azure/azure_rm_keyvault_facts.py

```diff
@@ -0,0 +1,193 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_keyvault_facts
+
+version_added: "2.5"
+
+short_description: Get key vault facts.
+
+description:
+    - Get facts for a specific key vault or all key vaults within a resource group.
+
+options:
+    resource_group:
+        description:
+            - Limit results by resource group. Required when filtering by name.
+    name:
+        description:
+            - Only show results for a specific key vault.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+    top:
+        description:
+            - Maximum number of results to return.
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Obezimnaka Boms @ozboms"
+
+'''
+
+EXAMPLES = '''
+    - name: Get facts for one key vault
+      azure_rm_keyvault_facts:
+        resource_group: Testing
+        name: foobar22
+
+    - name: Get facts for all zones in a resource group
+      azure_rm_keyvault_facts:
+        resource_group: Testing
+        top: 50
+
+    - name: Get facts by tags
+      azure_rm_keyvault_facts:
+        tags:
+          - testing
+'''
+
+RETURN = '''
+azure_keyvault:
+    description: List of key vault dicts.
+    returned: always
+    type: list
+    example:  []
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.common import AzureMissingResourceHttpError, AzureHttpError
+    from azure.mgmt.keyvault.models import ResourcePaged
+except:
+    # This is handled in azure_rm_common
+    pass
+
+AZURE_OBJECT_CLASS = 'KeyVault'
+
+
+class AzureRMKeyVaultFacts(AzureRMModuleBase):
+
+    def __init__(self):
+
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            name=dict(type='str'),
+            resource_group=dict(type='str'),
+            tags=dict(type='list'),
+            top=dict(type='int')
+        )
+
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_keyvault=[])
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.tags = None
+        self.top = None
+
+        super(AzureRMKeyVaultFacts, self).__init__(self.module_arg_spec)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        if self.name and not self.resource_group:
+            self.fail("Parameter error: resource group required when filtering by name.")
```

Comment:
Add check to spec, no need to check here

Reply:
done